image_segmentation.py

```diff
@@ -15,6 +15,8 @@
 import cv2
 import sys
 import numpy as np
+import matplotlib
+matplotlib.use("agg")    # must select backend before importing pyplot
```

Comment:
Setting the backend in code [is discouraged](https://matplotlib.org/faq/usage_faq.html#what-is-a-backend).

If this it to fix the CI, I would suggest setting `MPLBACKEND` env var instead.

The main reason being interference with a user-preferred backend. And at some point with the `leapide` backend (when we develop it).

Reply:
Hmm, unfortunately it's not just to fix the CI. Running `python image_segmentation.py` on python 3.5 (osx) will also give an import error if the backend isn't set. 

This is the error that I get:
```
from matplotlib.backends import _macosx
ImportError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends.
```

On a related note, there are also several other examples that are setting the backend to "agg" directly in the code.

Reply:
Luckily, this will be sorted out when we drop support for py35 (very soon!).

Reply:
In the meantime, since we already have some examples with hardcoded backends, I guess we can do it here as well...